interfaces/builtin/mount_control_test.go

```diff
@@ -0,0 +1,283 @@
+		},
+		{
+			"mount:\n  - what: /\n    where: /\n    type: [true, false]",
+			`mount-control "type" element 1 not a string.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n    type: [auto)]",
+			`mount-control filesystem type invalid.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n    type: [upperCase]",
+			`mount-control filesystem type invalid.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n    type: [two words]",
+			`mount-control filesystem type invalid.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n",
+			`mount-control "options" cannot be empty`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /\n    options: string",
+			`mount-control "options" must be an array of strings.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n    options: []",
+			`mount-control "options" cannot be empty`,
+		},
+		{
+			"mount:\n  - what: here\n    where: /",
+			`mount-control "what" attribute is invalid: must start with / and not contain special characters`,
+		},
+		{
+			"mount:\n  - what: /double\"quote\n    where: /",
+			`mount-control "what" attribute is invalid: must start with / and not contain special characters`,
+		},
+		{
+			"mount:\n  - what: /variables/are/not/@{allowed}\n    where: /",
+			`mount-control "what" attribute is invalid: must start with / and not contain special characters`,
+		},
+		{
+			"mount:\n  - what: /invalid}pattern\n    where: /",
+			`mount-control "what" setting cannot be used: Invalid closing brace, no matching open.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /path",
+			`mount-control "where" attribute is invalid: must start with \$SNAP_COMMON, \$SNAP_DATA or /media and not contain special characters`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/no\"quotes",
+			`mount-control "where" attribute is invalid: must start with \$SNAP_COMMON, \$SNAP_DATA or /media and not contain special characters`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/no@{variables}",
+			`mount-control "where" attribute is invalid: must start with \$SNAP_COMMON, \$SNAP_DATA or /media and not contain special characters`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/invalid[path",
+			`mount-control "where" setting cannot be used: Missing closing bracket ']'.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n    options: [sync,invalid]",
+			`mount-control option unrecognized or forbidden: "invalid"`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n    type: [ext4,debugfs]",
+			`mount-control forbidden filesystem type: "debugfs"`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/*\n    type: [ext4]\n    options: [rw,bind]",
+			`mount-control option "bind" is incompatible with a FS type declaration`,
+		},
+		{
+			"mount:\n  - what: /tmp/..\n    where: /media/*",
+			`mount-control "what" pattern is not clean:.*`,
+		},
+		{
+			"mount:\n  - what: /\n    where: /media/../etc",
```

Comment:
we should also have tests for `$SNAP_DATA/$SNAP_DATA` and `/$SNAP_DATA`, since we don't want to allow `$` again after the first variable instance AFAIK and only allow SNAP_DATA/SNAP_COMMON to be used when they are at the very start of the string

Reply:
Added